packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -109,12 +128,211 @@ const mapCertificate = (certificate: Schema.Certificate): Cardano.Certificate =>
+    coins: txOut.value.coins
+  }
+});
+
+const mapMint = (tx: Schema.TxMary): Cardano.TokenMap | undefined => {
+  if (tx.body.mint.assets === undefined) return undefined;
+  return new Map(Object.entries(tx.body.mint.assets).map(([key, value]) => [Cardano.AssetId(key), value]));
+};
+
+const mapScriptIntegrityHash = ({
+  body: { scriptIntegrityHash }
+}: Schema.TxAlonzo): Cardano.util.Hash32ByteBase16 | undefined => {
+  if (scriptIntegrityHash === null) return undefined;
+  return Cardano.util.Hash32ByteBase16(scriptIntegrityHash);
+};
+
+const mapValidityInterval = ({
+  invalidBefore,
+  invalidHereafter
+}: Schema.ValidityInterval): Cardano.ValidityInterval => ({
+  invalidBefore: invalidBefore ? Cardano.Slot(invalidBefore) : undefined,
+  invalidHereafter: invalidHereafter ? Cardano.Slot(invalidHereafter) : undefined
+});
+
+const mapCommonTx = (tx: CommonBlock['body'][0], kind: BlockKind): Cardano.Tx => ({
+  auxiliaryData: mapAuxiliaryData(tx.metadata),
+  body: {
+    certificates: tx.body.certificates.map(mapCertificate),
+    collaterals: isAlonzoOrAbove(kind) ? (tx as Schema.TxAlonzo).body.collaterals.map(mapTxIn) : undefined,
+    fee: tx.body.fee,
+    inputs: tx.body.inputs.map(mapTxIn),
+    mint: isMaryOrAbove(kind) ? mapMint(tx as Schema.TxMary) : undefined,
+    outputs: tx.body.outputs.map(mapTxOut),
+    requiredExtraSignatures: isAlonzoOrAbove(kind)
+      ? (tx as Schema.TxAlonzo).body.requiredExtraSignatures.map(Cardano.Ed25519KeyHash)
+      : undefined,
+    scriptIntegrityHash: isAlonzoOrAbove(kind) ? mapScriptIntegrityHash(tx as Schema.TxAlonzo) : undefined,
+    validityInterval: isShelleyTx(kind)
+      ? undefined
+      : mapValidityInterval((tx as Schema.TxAlonzo).body.validityInterval),
+    withdrawals: Object.entries(tx.body.withdrawals).map(([key, value]) => ({
+      quantity: value,
+      stakeAddress: Cardano.RewardAccount(key)
+    }))
+  },
+  id: Cardano.TransactionId(tx.id),
+  witness: {
+    bootstrap: tx.witness.bootstrap.map(mapBootstrapWitness),
+    datums: isAlonzoOrAbove(kind)
+      ? Object.values((tx as Schema.TxAlonzo).witness.datums).map((d) => Cardano.util.HexBlob(d))
+      : undefined,
+    redeemers: isAlonzoOrAbove(kind)
+      ? Object.entries((tx as Schema.TxAlonzo).witness.redeemers).map(([key, value]) => mapRedeemer(key, value))
+      : undefined,
+    scripts: [...Object.values(tx.witness.scripts).map(mapScript)],
+    signatures: new Map(
+      Object.entries(tx.witness.signatures).map(([key, value]) => [
+        Cardano.Ed25519PublicKey(key),
+        Cardano.Ed25519Signature(Cardano.util.HexBlob.fromBase64(value).toString())
+      ])
+    )
+  }
+});
+
+export const mapCommonBlockBody = ({ body }: CommonBlock, kind: BlockKind): Cardano.Block['body'] =>
+  body.map((blockBody) => mapCommonTx(blockBody, kind));
+
+const mapByronTx = (tx: Schema.TxByron): Cardano.Tx => ({
+  body: {
+    fee: 0n, // Not sure how to calculate the fee from ByronTx there is no fee?
```

Comment:
[TxFeePolicy](https://github.com/input-output-hk/cardano-js-sdk/blob/f1cebb4d795847e3f992d1f6869dcac47842d5e6/packages/core/src/Cardano/types/ProtocolParameters.ts#L33-L36) from the [Byron protocol params](https://github.com/input-output-hk/cardano-js-sdk/blob/f1cebb4d795847e3f992d1f6869dcac47842d5e6/packages/core/src/Cardano/types/ProtocolParameters.ts#L57).

```
coefficient x tx size + constant

Reply:
Since `TxFreePolicy` is part of `ProtocolParametersByron`, but in the [mapping](https://github.com/input-output-hk/cardano-js-sdk/blob/5f231dada0a71522cc6ed5c61375d7cdbeb65393/packages/ogmios/src/ogmiosToCore/block.ts#L175-L176), we have access only to the Ogmios block, how do we prefer to get the byron protocol params? 
Could we extend the `const block = (ogmiosBlock: Schema.Block): Cardano.Block` mapping with another arg: `params: ProtocolParametersByron`? @rhyslbw

Reply:
Yes, that was my thought

Reply:
How about getting tx size? Something like `txPayload

Reply:
try `Buffer.from(raw, 'hex').length`

Reply:
Added with ad5ba61def0fa35d358a03810152f2c068dbf801